tox.ini

```diff
@@ -0,0 +1,47 @@
+# this file is *not* meant to cover or endorse the use of tox or pytest or testing in general,
```

Comment:
what is this file for?

Reply:
it is a configuration of testing and defining test and formatting configurations at one place...
see https://tox.readthedocs.io/en/latest